nixos/modules/services/web-apps/jitsi-meet.nix

```diff
@@ -0,0 +1,497 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.jitsi-meet;
+  dataDir = "/var/lib/jitsi-meet";
+  attrsToArgs = a: concatStringsSep " " (mapAttrsToList (k: v: "${k}=${toString v}") a);
+
+  # HOCON is a JSON superset that videobridge2 uses for configuration
+  # it can substitute environment variables which we use for passwords here
+  # https://github.com/lightbend/config/blob/master/README.md
+  toHOCON = v: builtins.replaceStrings [ ''"__ENV__'' ''__ENV__"'' ] [ "\${" "}" ] (builtins.toJSON v);
+  jvbArgs = {
+    "--apis" = "none";
+  };
+  jvbProps = {
+    "-Dnet.java.sip.communicator.SC_HOME_DIR_LOCATION" = "/etc/jitsi";
+    "-Dnet.java.sip.communicator.SC_HOME_DIR_NAME" = "videobridge";
+    "-Djava.util.logging.config.file" = "/etc/jitsi/videobridge/logging.properties";
+    "-Dconfig.file" = pkgs.writeText "jvb.conf" (toHOCON jvbConfig);
+  } // (mapAttrs' (k: v: nameValuePair "-D${k}" v) cfg.videobridge.extraProperties);
+  jvbConfig = recursiveUpdate defaultJvbConfig cfg.videobridge.config;
+
+  jicofoArgs = {
+    "--host" = "localhost";
+    "--domain" = cfg.hostName;
+    "--secret" = "\${JICOFO_COMPONENT_SECRET}";
+    "--user_domain" = "auth.${cfg.hostName}";
+    "--user_name" = "focus";
+    "--user_password" = "\${JICOFO_USER_SECRET}";
+  };
+  jicofoProps = {
+    "-Dnet.java.sip.communicator.SC_HOME_DIR_LOCATION" = "/etc/jitsi";
+    "-Dnet.java.sip.communicator.SC_HOME_DIR_NAME" = "jicofo";
+    "-Djava.util.logging.config.file" = "/etc/jitsi/jicofo/logging.properties";
+    "-Djavax.net.ssl.trustStore" = "${dataDir}/prosody-ca.jks";
+    "-Djavax.net.ssl.trustStorePassword" = "javapls";
+  };
+  jicofoConfig = pkgs.writeText "sip-communicator.properties" (
+    concatStringsSep "\n" (mapAttrsToList (k: v: "${k}=${v}") cfg.jicofo.config)
+  );
```

Comment:
I'd prefer if these args could all be a bit more closer to their usage - they're not used in multiple places anyways.